beets/dbcore/query.py

```diff
@@ -257,25 +264,26 @@ def string_match(cls, pattern: str, value: str) -> bool:
         return pattern.lower() in value.lower()
 
 
-class RegexpQuery(StringFieldQuery):
+class RegexpQuery(StringFieldQuery[Pattern]):
     """A query that matches a regular expression in a specific Model field.
 
     Raises InvalidQueryError when the pattern is not a valid regular
     expression.
     """
 
     def __init__(self, field: str, pattern: str, fast: bool = True):
-        super().__init__(field, pattern, fast)
         pattern = self._normalize(pattern)
         try:
-            self.pattern = re.compile(self.pattern)
+            pattern_re = re.compile(pattern)
         except re.error as exc:
             # Invalid regular expression.
             raise InvalidQueryArgumentValueError(pattern,
                                                  "a regular expression",
                                                  format(exc))
 
-    def col_clause(self):
+        super().__init__(field, pattern_re, fast)
```

Comment:
[non-actionable] The diff confused me for a sec here, so for future reviewers: here's where the `super` call moved to.